File: payment-service/src/main/java/com/stackroute/service/PaymentService.java
package com.stackroute.service;

import com.razorpay.RazorpayException;
import com.stackroute.exception.OrderIdNotFoundException;
import com.stackroute.model.PaymentRequest;
import com.stackroute.model.PaymentSuccess;

public interface PaymentService {
    String createOrder(PaymentRequest paymentRequest) throws RazorpayException;

    String savePaymentId(PaymentSuccess paymentSuccess) throws OrderIdNotFoundException;
}

File: product-webapp/angularbase/carnivry/src/app/service/payment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { RazorpayOrder } from '../model/razorpay-order';
import { RazorpaySuccess } from '../model/razorpay-success';

@Injectable({
  providedIn: 'root',
})
export class PaymentService {
  constructor(private myClient: HttpClient) {}


  // baseUrl= environment.baseUrl+"payment/api/v1";
  private controllerUrl = '/api/v1';
  private paymentUrl = environment.baseUrl + '/payment' + this.controllerUrl;


  createOrder(data: any): Observable<RazorpayOrder> {
    return this.myClient.post<RazorpayOrder>(
      this.paymentUrl + '/create_order',
      data,
      { responseType: 'json' }
    );
  }

  paymentSuccess(data: any) {
    return this.myClient.post(this.paymentUrl + '/payment_success', data, {
      responseType: 'text',
    });
  }
}
